### ISSUE TITLE
More Configuration Params for Automation

### ISSUE THREAD

**Issue by designermonkey** [swarmpit/swarmpit#641]
I really like the slick and easy use of Swarmpit, but it isn't very useable in automation.

Our automation requirements mean that we would need to set a user up automatically when the image is deployed, and also a registry too.

I understand that the registry cannot be added without a user (I read another closed issue), so I would like to ask if it were possible to add a couple more ENV vars to allow automation.

Specifically `SWARMPIT_USER`, `SWARMPIT_PASSWORD` for starters to automatically create a default user account.

After that, some ENV vars for creating a registry would be wonderful to allow us to automate the deployment of Swarmpit.

**Comment by nohaapav** [comment]
Hi @designermonkey, you can create user simply with API call. See swarmpit swagger doc:
<img width="970" alt="Screen Shot 2022-05-27 at 2 08 53 PM" src="https://user-images.githubusercontent.com/4179104/170696594-0fddb264-c2c0-4de9-b75c-22bff5c19601.png">

**Comment by nohaapav** [comment]
The same way you can create registry: Just check with inspector what data are needed in payload. E.g. for dockerhub:

`POST /api/registry/dockerhub`

```
{"username":"user","password":"pass","public":false}
```

**Comment by designermonkey** [comment]
@optica-phoffmann thanks for pointing that out, I hadn't got to replying yet as I've been on holiday but this was my concern as well.

@nohaapav there needs to be some way of passing variables over the environment for use in docker for example. Even if it were as simple as an `API_TOKEN` variable.

**Comment by designermonkey** [comment]
I don't mean to be rude, but can I bump this issue for a response?

**Comment by nohaapav** [comment]
> > You just have to generate token first in API access section.
> 
> @nohaapav Could you please elaborate, how I can create the initial admin user via API? As far as I can see all API commands require authorization. To authenticate I need to create an admin user via UI first.

Yes first user must be created manually .. or you can create admin user by REST call in influx DB ;)

**Comment by designermonkey** [comment]
So we've gone full circle back to the initial question. Can we have a way added to the service where we can create the first user automatically?

**Comment by nohaapav** [comment]
See what command swarmpit is using on init .. We prompt user to enter credentials which we use to bootstrap admin acc. You can follow the same pattern.

**Comment by Adam7288** [comment]
Why was this removed after version 1.9? It is very useful.
